### DISCUSSION TITLE
Sorting habits by score in reverse order

### DISCUSSION THREAD

**Discussion by macpac59** [iSoron/uhabits#497] [Feature Requests] [answered]
Hi there,

First of all: a simple but great app! I'm just sorry for all the iOS users who can't use it...

It would be great to offer an option (maybe a check box or a sub menu) to sort the habits by score but in reverse order. This way, the weakest habit would be on top and thus in one's focus on what to work on primarily.

Best regards
macpac59

**Comment by ComedyTomedy** [comment]
+1 for this

**Comment by sanjaymaniam** [comment]
Yup, this would be useful.

**Comment by scrimbly** [comment]
I found myself wanting this too. I hacked up a quick fix that just adds another menu item, but I'm curious what UX the community would prefer. I can imagine a few options, but I'm open to more:

1. Two menu items, different icon
    - By score (V)
    - By score (^)
2. Two menu items, different text
    - By score desc
    - By score asc
3. One menu item, toggles invisibly
    - By score
4. One menu item, toggles icon
    - By score (V)
5. One menu item, toggles text
    - By score (asc)

And if we choose a text option, what (English) text do we want?
- descending, ascending
- desc, asc
- (desc), (asc)
- (hi first), (low first)
- By highest score, By lowest score

**Comment by macpac59** [comment]
Cool that you take care of this topic! I would perceive solutions (1) and
(4) as quite intuitive. Texts get too long and (3) might be confusing.


Am Do., 13. Feb. 2020 um 03:33 Uhr schrieb scrimbly <
notifications@github.com>:

**Comment by scrimbly** [comment]
I tried adding :arrow_up_small: and :arrow_down_small: icons to the menu items, and it looks like this:

![sort-menu](https://user-images.githubusercontent.com/38977720/74901426-72720280-5368-11ea-960d-2f25b498c7a9.png)

**Reply by naught101** [reply]
After would be more readable

**Comment by macpac59** [comment]
Looks nice! Will you soon implement this function in the main app or create
a fork?

scrimbly <notifications@github.com> schrieb am Do., 20. Feb. 2020, 05:44:

**Comment by scrimbly** [comment]
@macpac59 I opened PR #556 and hope for it to be merged into the main app once @iSoron approves it

**Comment by iSoron** [comment]
Thank you for the pull request, @scrimbly. I was away from the project for a few weeks, but I am planning to review it soon. Many users would certainly find this feature useful; it has been requested a couple of times.

**Comment by macpac59** [comment]
Cool, when will this be published?

Am Sa., 15. Aug. 2020 um 19:30 Uhr schrieb Álinson S. Xavier <
notifications@github.com>:

**Comment by iSoron** [comment, marked as answer]
This feature has been implemented and will be available in Loop 2.0.
